osquery/tables/system/windows/logged_in_users.cpp

```diff
@@ -108,16 +110,35 @@ QueryData genLoggedInUsers(QueryContext& context) {
     }
 
     r["pid"] = INTEGER(-1);
-    results.push_back(r);
 
     if (clientInfo != nullptr) {
       WTSFreeMemoryEx(WTSTypeSessionInfoLevel1, clientInfo, count);
       clientInfo = nullptr;
+      wtsClient = nullptr;
     }
+
+    const auto sidBuf =
+        getSidFromUsername(stringToWstring(wtsSession->UserName));
+
     if (sessionInfo != nullptr) {
       WTSFreeMemoryEx(WTSTypeSessionInfoLevel1, sessionInfo, count);
       sessionInfo = nullptr;
+      wtsSession = nullptr;
     }
+
+    if (sidBuf == nullptr) {
+      VLOG(1) << "Error converting username to SID";
+      results.push_back(r);
+      continue;
+    }
+
+    const auto sidStr = psidToString(reinterpret_cast<SID*>(sidBuf.get()));
+    r["sid"] = SQL_TEXT(sidStr);
+
+    const auto hivePath = "HKEY_USERS\\" + sidStr;
```

Comment:
Small question, do we need that constant prefix for path? I mean, if it is always the same what sort of sense does it make?

Reply:
Why do we need hive at all. If it's always possible to determine it just based on sid, why not just add the sid?

Reply:
My reasoning here is that having the full hive path will make it easier to use the results of this table in subqueries, especially for (currently hypothetical) tables that take a full hive context as a `where` clause.

A contrived example:

```sql
SELECT * FROM user_config WHERE hive = (SELECT registry_hive FROM logged_in_users user = "foobar")
```

Thoughts? I'm aware that it's a bit of a duplication; if that's a hard blocker, I'll remove it. If not, I'd like to include it: it would make pivoting around the registry from within osquery a bit easier 🙂

Reply:
An actual functional example: selecting registry keys for active users:

```sql
select * from registry where key like (select registry_hive from logged_in_users where type = "active");
```

Reply:
It's good in terms of usability, so let's keep it.